test/system/700-play.bats

```diff
@@ -386,3 +386,27 @@ status: {}
     run_podman rm -a -f
     run_podman rm -f -t0 myyaml
 }
+
+@test "podman kube play - hostport" {
+    HOST_PORT=$(random_free_port)
+    echo "
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    app: test
+  name: test_pod
+spec:
+  containers:
+    - name: server
+      image: $IMAGE
+      ports:
+        - name: hostp
+          containerPort: $HOST_PORT
+" > $PODMAN_TMPDIR/testpod.yaml
+
+    run_podman kube play $PODMAN_TMPDIR/testpod.yaml
+    run_podman pod inspect test_pod --format "{{.InfraConfig.PortBindings}}"
+    is "$output" "map[$HOST_PORT/tcp:[{ $HOST_PORT}]]"
```

Comment:
Whew! Figured out why test passes on `main`. It's because the `is` function tries an equal-match (==) first, but then tries `expr` if that fails. Which means all those brackets end up meaning "map, plus maybe other stuff". Fix:
```diff
-    is "$output" "map[$HOST_PORT/tcp:[{ $HOST_PORT}]]"
+    assert "$output" = "map[$HOST_PORT/tcp:[{ $HOST_PORT}]]"
```
`asserts`, with explicit `=`, will not try a regex match. With this fix, the test fails on main and passes with your PR.